packages/datadog-instrumentations/src/fetch.js

```diff
@@ -0,0 +1,46 @@
+'use strict'
+
+const shimmer = require('../../datadog-shimmer')
+const { channel } = require('./helpers/instrument')
+
+const startChannel = channel('apm:fetch:request:start')
+const finishChannel = channel('apm:fetch:request:finish')
+const errorChannel = channel('apm:fetch:request:error')
+
+function wrapFetch (fetch, Request) {
+  if (typeof fetch !== 'function') return fetch
+
+  return function (input, init) {
+    if (!startChannel.hasSubscribers) return fetch.apply(this, arguments)
+
+    const req = new Request(input, init)
+    const headers = req.headers
+    const message = { req, headers }
+
+    startChannel.publish(message)
+
+    // Request object is read-only so we need new objects to change headers.
+    arguments[0] = message.req
+    arguments[1] = { headers: message.headers }
+
+    return fetch.apply(this, arguments)
+      .then(
+        res => {
+          finishChannel.publish({ req, res })
```

Comment:
Do you not have to return the `res` here?

Reply:
Yeah, your tests never validate the return value (or rather, the resolved value) of `fetch()`, nor rejections, as far as I can tell. (Though you pass the rejection through by throwing it, it looks like.

Reply:
Rejection is tested through the error tests, but the response was indeed not tested and not returned. I pushed a fix.